test/integration/Client/client_lib_thread_interdependency_test.cpp

```diff
@@ -416,19 +455,52 @@ int main(int argc, char**argv)
         return -1;
     }
 
+    // Parse number of threads from command line, default to DEFAULT_NUM_THREADS
+    int num_threads = parse_num_threads_arg(argc, argv);
+
     // Initiate test
-    LOG_INFO("[ClientLibThreadInterdependencyTest] Running test.");
-    int num_threads = 8;
-    std::vector <struct thread_arg> t_args(num_threads);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Running test with " + std::to_string(num_threads) + " threads.");
+    t_args.resize(num_threads);
     std::vector <std::thread> workers(num_threads);
     shared_state.resize(num_threads, 0);
 
+    // Randomization setup
+    std::random_device rd;
+    std::mt19937 gen(rd());
+
+    // Generate random assignments for story-to-thread
+    std::uniform_int_distribution <> story_dis(0, 3); // Range [0, 3] for 4 stories: 11, 12, 21, 22
+    std::vector <int> story_assignments(num_threads);
+    for(int i = 0; i < num_threads; ++i)
+    {
+        story_assignments[i] = story_dis(gen);
+    }
+
+    // Generate random sleep times
+    const int max_sleep_time = 10; // Define maximum sleep time in seconds
+    std::uniform_int_distribution <> sleep_dis(0, max_sleep_time);
+    std::vector <int> sleep_times(num_threads);
+    for(int i = 0; i < num_threads; ++i)
+    {
+        sleep_times[i] = sleep_dis(gen);
+    }
+
+    // Chronicle and story mapping
+    const std::vector <std::string> story_mapping = {"11", "12", "21", "22"};
+
     // Create and start the worker threads
     for(int i = 0; i < num_threads; i++)
     {
-        t_args[i].tid = i;  // Assign thread ID
-        std::thread t{thread_body, &t_args[i]};  // Start the thread
-        workers[i] = std::move(t);  // Move thread to workers vector
+        t_args[i].tid = i; // Assign thread ID
+        t_args[i].chronicle_name =
+                "CHRONICLE_" + story_mapping[story_assignments[i]].substr(0, 1); // Extract chronicle digit
+        t_args[i].story_name = "STORY_" + story_mapping[story_assignments[i]].substr(1, 1);       // Extract story digit
+
+        // Simulate sleep before task begins
+        std::this_thread::sleep_for(std::chrono::seconds(sleep_times[i]));
```

Comment:
Do we really need this long sleep for all threads? When `num_threads` is high, it may take too long to just start all threads.

Reply:
sleep_times[i] value for each thread should be passed into the thread as another thread argument.
this is the number of seconds that each thread sleeps at its start. 
this way the sleeping is parallel , not sequential

Reply:
You are right